src/controllers/auth.controller.ts

```diff
@@ -5,33 +5,51 @@ import type Profile from '../entities/profile.entity'
 import jwt from 'jsonwebtoken'
 import { JWT_SECRET } from '../configs/envConfig'
 
-export const register = async (req: Request, res: Response): Promise<void> => {
+interface AuthResponse {
```

Comment:
Hi @Shrenik0321, Instead of creating interfaces for each response type we can use typescript generics here,

```
interface ApiResponse<T> {
  statusCode: number;
  message?: string;
  data?: T | null;
}

// For CategoryResponse
const categoryResponse: ApiResponse<Category> = {
  statusCode: 200,
  message: "Success",
  data: someCategoryObject,
};

// For MentorStatusResponse
const mentorResponse: ApiResponse<Mentor> = {
  statusCode: 200,
  message: "Success",
  data: someMentorObject,
};

// For AuthResponse
const authResponse: ApiResponse<Profile> = {
  statusCode: 200,
  message: "Success",
  data: someProfileObject,
};

```

Reply:
![image](https://github.com/sef-global/scholarx-backend/assets/92735170/a23cc666-9aa0-4d85-911e-bac140f6bf01)


is this right ? just for reconfirmation @anjula-sack

Reply:
yes move the ApiResponse interface to types file and import them accordingly